app/components/ImageUpload.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <button
+    type="button"
+    :class="$style.component"
+    @click="onButtonClick"
+  >
+    <img
+      v-if="imageFileObjectUrl"
+      :src="imageFileObjectUrl"
+      alt="Uploaded image"
+      :class="$style.image"
+    />
+
+    <Icon
+      v-else
+      size="1.5rem"
+      name="tabler:upload"
+      :class="$style.icon"
+    />
+
+    <input
+      type="file"
+      accept="image/png, image/jpeg, image/webp"
+      ref="fileInput"
+      :class="$style.input"
+      @change="onFileChange"
+    />
+  </button>
+</template>
+
+<script lang="ts" setup>
+  const fileInputRef = useTemplateRef('fileInput');
+  const imageFile = ref<File | null>(null);
+
+  const imageFileObjectUrl = computed(() => {
+    if (imageFile.value === null) {
+      return null;
+    }
+
+    return URL.createObjectURL(imageFile.value);
```

Comment:
You should remove the object url when `imageFile` is changed.

https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static